source/opt/mem_pass.cpp

```diff
@@ -873,5 +850,32 @@ bool MemPass::CFGCleanup(ir::Function* func) {
   return modified;
 }
 
+void MemPass::CollectTargetVars(ir::Function* func) {
+  seen_target_vars_.clear();
+  seen_non_target_vars_.clear();
+  supported_ref_vars_.clear();
+  type2undefs_.clear();
+
+  // Collect target (and non-) variable sets. Remove variables with
+  // non-load/store refs from target variable set
+  for (auto& blk : *func) {
```

Comment:
Open a follow up issue for this.

Reply:
Not only variables need to be local storage.  This collector also makes sure that they are accessed without any decorations.  This code predates me.  I needed to incorporate it from the old Phi inserter to avoid the random segfaults I was getting before.  Not every load/store to local functions can be converted into SSA ids.

This loop does not even show up in profiles that I've seen, so it's not a big concern.